kernel/core/maths/functionimplementation.cpp

```diff
@@ -2737,6 +2737,77 @@ value_type parser_isnan(const value_type& v)
 }
 
 
+/////////////////////////////////////////////////
+/// \brief This function returns the utc offset
+///
+/// \return value_type
+///
+/////////////////////////////////////////////////
+value_type parser_get_utc_offset()
+{
+    time_zone tz = getCurrentTimeZone();
+    int utc_offset = tz.Bias.count() + tz.DayLightBias.count();
+    // return values in seconds
+    return -1 * utc_offset * 60;
+}
+
+
+/////////////////////////////////////////////////
+/// \brief This function returns whether the year
+/// is a leap year
+///
+/// \param nYear const value_type&
+/// \return value_type
+///
+/////////////////////////////////////////////////
+value_type parser_is_leap_year(const value_type& nDate)
+{
+    time_stamp ts = getTimeStampFromTimePoint(to_timePoint(nDate.real()));
+    int nYear = int(ts.m_ymd.year());
+
+    if (intCast(nYear) % 4 == 0){
+        if (intCast(nYear) % 100 != 0 || intCast(nYear) % 400 == 0) {
+            return 1.0;
+        }
+    }
+    return 0.0;
+}
+
+
+
+/////////////////////////////////////////////////
+/// \brief This function returns whether the
+/// current time is a summer time
+///
+/// \param vElements const value_type*
+/// \param nElements int
+/// \return value_type
+///
+/////////////////////////////////////////////////
+value_type parser_is_daylightsavingtime(const value_type& nDate)
+{
+    time_stamp ts = getTimeStampFromTimePoint(to_timePoint(nDate.real()));
+    int year = int(ts.m_ymd.year());
+    int day = unsigned(ts.m_ymd.day());
+    int month = unsigned(ts.m_ymd.month());
+
+    std::tm time_info = {};
+    time_info.tm_year = year - 1900;
+    time_info.tm_mon = month - 1;
+    time_info.tm_mday = day;
+    time_info.tm_hour = 12;          // Set to noon to avoid ambiguity
+
+    std::time_t timestamp = std::mktime(&time_info);
+    std::tm* local_time = std::localtime(&timestamp);

```

Comment:
Please check, whether `__time64_t` and `_mktime64()` and `_localtime64()` are available in your compiler (might not be found by Code::Blocks, but still compiles). The first one should be, because you're using that already.